util/attendance/src/lib.rs

```diff
@@ -0,0 +1,81 @@
+#![no_std]
+
+#[allow(unused_imports)]
+use multiversx_sc::imports::*;
+
+#[multiversx_sc::contract]
+pub trait Attendance: multiversx_sc_modules::only_admin::OnlyAdminModule {
+    #[init]
+    fn init(&self, admins: MultiValueEncoded<ManagedAddress>) {
+        self.admins().extend(admins);
+    }
+
+    #[only_owner]
+    #[upgrade]
+    fn upgrade(&self) {}
+
+    #[endpoint(registerStudent)]
+    fn register_student(&self) {
+        let caller = self.blockchain().get_caller();
+
+        require!(
+            self.students().contains(&caller),
+            "Student is already registered"
+        );
+
+        self.students().insert(caller);
+    }
+
+    #[endpoint(registerAttendance)]
+    fn register_attendance(&self) {
+        let caller = self.blockchain().get_caller();
+        self.require_caller_registered(&caller);
+
+        self.attendance(&caller)
```

Comment:
Only if `secret` is provided and it's correct.

Reply:
please take a look at the logic for the epoch, at the moment i put the `current_block_round` + 10